web/src/features/map/Map.tsx

```diff
@@ -371,64 +361,14 @@ export default function MapPage({ onMapLoad }: MapPageProps): ReactElement {
       style={{ minWidth: '100vw', height: '100vh' }}
       mapStyle={MAP_STYLE as mapboxgl.Style}
     >
-      <Layer id="ocean" type="background" paint={styles.ocean} />
-
-      <Source
-        id="zones-clickable"
-        promoteId={'zoneId'}
-        type="geojson"
-        data={worldGeometries}
-      >
-        <Layer id="zones-clickable-layer" type="fill" paint={styles.zonesClickable} />
-        <Layer id="zones-hoverable-layer" type="fill" paint={styles.zonesHover} />
-        <Layer id="zones-border" type="line" paint={styles.zonesBorder} />
+      <Layer id="ocean" type="background" paint={mapStyles.ocean} />
+      <Source id={ZONE_SOURCE} promoteId={'zoneId'} type="geojson" data={worldGeometries}>
+        <Layer id="zones-clickable-layer" type="fill" paint={mapStyles.zonesClickable} />
+        <Layer id="zones-hoverable-layer" type="fill" paint={mapStyles.zonesHover} />
+        <Layer id="zones-border" type="line" paint={mapStyles.zonesBorder} />
       </Source>
       <Source id="states" type="geojson" data={statesGeometries}>
-        <Layer id="states-border" type="line" paint={styles.statesBorder} minzoom={2.5} />
-        <Layer
-          id="state-labels-name"
-          type="symbol"
-          source="states"
-          layout={{
-            'text-field': ['get', 'stateName'],
-            'symbol-placement': 'point',
-            'text-size': 12,
-            'text-letter-spacing': 0.12,
-            'text-transform': 'uppercase',
-
-            'text-font': ['poppins-semibold'],
-          }}
-          paint={{
-            'text-color': 'white',
-            'text-halo-color': '#111827',
-            'text-halo-width': 0.5,
-            'text-halo-blur': 0.25,
-            'text-opacity': 0.9,
-          }}
-          minzoom={4.5}
-        />
-        <Layer
-          id="state-labels-id"
-          type="symbol"
-          source="states"
-          layout={{
-            'text-field': ['get', 'stateId'],
-            'symbol-placement': 'point',
-            'text-size': 12,
-            'text-letter-spacing': 0.12,
-            'text-transform': 'uppercase',
-            'text-font': ['poppins-semibold'],
-          }}
-          paint={{
-            'text-color': 'white',
-            'text-halo-color': '#111827',
-            'text-halo-width': 0.5,
-            'text-halo-blur': 0.25,
-            'text-opacity': 0.9,
-          }}
-          maxzoom={4.5}
-          minzoom={3}
-        />
+        <StatesLayer mapStyles={mapStyles} />
```

Comment:
Makes good sense to extract this to it's own component/file. How about also splitting "mapStyles" and moving the relevant styling to the new file as well?
And could we move `<Source />` too, so everything related to states is co-living in the same place? :)

Reply:
Good point ✅